prow/test/integration/test/moonraker_test.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	v1 "k8s.io/api/core/v1"
+
+	prowjobv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/moonraker"
+	"k8s.io/test-infra/prow/test/integration/internal/fakegitserver"
+)
+
```

Comment:
Originally I was thinking neither the unit nor integration tests were properly validating that Moonraker returns the correct ProwJob YAML for different PRs. However, this appears to be well tested in https://github.com/kubernetes/test-infra/blob/master/prow/config/cache_test.go and Moonraker's serve function is a simple marshal/unmarshal wrapper so this integration test seems sufficient. :+1:

Reply:
Exactly.